### PR TITLE
Use api total consumption and product values

### PR DESCRIPTION
**Pull request by tonypls**
## Issue
Use total values for emissions and electricity

## Description
This PR updates the app to use the V7 zone data to display total emissions and total electricity. 

I still have to test a few things but early feedback is very much welcomed!

Dependent on release of [this PR](https://github.com/electricitymaps/electricitymaps/pull/3939)

The total values use this assumption taken from flowtracing:

```

row["total_consumption_avg"] = (

total_production_avg

+ (total_import_avg or 0)

- (total_export_avg or 0)

- (total_storage_avg or 0)

+ (total_discharge_avg or 0)

)
```

Note: Should update naming to not be hour specific

### CODE REVIEW

**Comment by tonypls** [comment]
I've tested the emission values and percentages for consumption and production in France and they add up to 100% for both ✅

File: web/src/features/charts/bar-breakdown/utils.ts [hunk 1]
```diff
@@ -61,15 +61,15 @@ export const getProductionData = (data: ZoneDetail): ProductionDataType[] =>
     const gCo2eqPerkWh = getProductionCo2Intensity(mode, data);
     const value = isStorage ? storage : production || 0;
     const gCo2eqPerHour = gCo2eqPerkWh * 1e3 * (value || 0);
-    const tCo2eqPerMin = gCo2eqPerHour / 1e6 / 60;
+    // const gCo2eqPerHour = gCo2eqPerHour / 1e6 / 60;
```
Comment (VIKTORVAV99) [inline web/src/features/charts/bar-breakdown/utils.ts:64]: Should we just remove this?

**Review by VIKTORVAV99 — COMMENTED** [review]
Just a few comments but otherwise things looks good despite the number of changes in here, good work!

File: web/src/features/charts/graphUtils.ts [hunk 2]
```diff
@@ -71,26 +71,38 @@ export const getGenerationTypeKey = (name: string): GenerationType | undefined =
...
   }
-  if (productionValue === null) {
+  if (zoneData.totalProduction === null) {
     return Number.NaN;
   }
+  // Electricity: Handle discharge case if available, else default to production
+  return zoneData.totalDischarge !== null && zoneData.totalDischarge !== undefined
+    ? zoneData.totalProduction + zoneData.totalDischarge
+    : zoneData.totalProduction;
+}
```
Comment (madsnedergaard) [inline web/src/features/charts/graphUtils.ts:89]: I think this can't be put more simply like this :)

```suggestion
return zoneData.totalProduction + (zoneData.totalDischarge II 0)
}
```

Comment (madsnedergaard) [reply web/src/features/charts/graphUtils.ts:89]: And same with emissions function below

File: web/src/features/charts/graphUtils.ts [hunk 3]
```diff
@@ -71,26 +71,38 @@ export const getGenerationTypeKey = (name: string): GenerationType | undefined =
...
-  const includeImports = mixMode === Mode.CONSUMPTION;
-  let productionValue: number;
-
-  if (displayByEmissions) {
-    productionValue = zoneData.totalCo2Production + zoneData.totalCo2Discharge;
-    productionValue += includeImports ? zoneData.totalCo2Import : 0;
-  } else {
-    productionValue = zoneData.totalProduction + zoneData.totalDischarge;
-    productionValue += includeImports ? zoneData.totalImport : 0;
+export function getTotalElectricity(zoneData: ZoneDetail, mixMode: Mode) {
```
Comment (madsnedergaard) [inline web/src/features/charts/graphUtils.ts:74]: Nice that you're splitting this function up, it was quite confusing that it did both! :)

File: web/src/utils/formatting.test.ts [hunk 4]
```diff
@@ -23,27 +23,42 @@ describe('formatCo2', () => {
     const expected = '0 t';
     expect(actual).toBe(expected);
   });
+  it('uses same unit as another value would non-zero', () => {
+    const actual = formatCo2(230_500, 2_350_000);
+    const expected = '0.2 t';
+    expect(actual).toBe(expected);
+  });
```
Comment (madsnedergaard) [inline web/src/utils/formatting.test.ts:30]: Not sure I understand this test, is it supposed to be more different from the one below?

File: web/src/utils/formatting.test.ts [hunk 5]
```diff
@@ -23,27 +23,42 @@ describe('formatCo2', () => {
...
     const expected = '0.2 t';
     expect(actual).toBe(expected);
   });
   it('handles values up to 100k', () => {
-    const actual = formatCo2(700_003_500_000);
-    const expected = '11,667 t';
+    const actual = formatCo2(99_000_000_000_000_000);
+    const expected = '99 Gt';
     expect(actual).toBe(expected);
   });
```
Comment (madsnedergaard) [inline web/src/utils/formatting.test.ts:41]: I'd like to keep a test for values up to 100k to test the thousand separator, was this changed to that high on purpose?

File: web/src/utils/formatting.test.ts [hunk 6]
```diff
@@ -23,27 +23,42 @@ describe('formatCo2', () => {
...
-  it('handles values up to 1 million', () => {
-    const actual = formatCo2(30_000_035_000_000);
-    const expected = '0.5 Mt';
+  it('handles values up to megatonnes', () => {
+    const actual = formatCo2(500_000_000_000_000);
+    const expected = '500 Mt';
     expect(actual).toBe(expected);
   });
```
Comment (madsnedergaard) [inline web/src/utils/formatting.test.ts:46]: The test name and expected value doesn't match :)

File: web/src/utils/formatting.test.ts [hunk 7]
```diff
@@ -23,27 +23,42 @@ describe('formatCo2', () => {
...
-  it('handles values above 1 million', () => {
-    const actual = formatCo2(400_000_350_000_000_000);
+  it('handles values up to 1 trillion grams', () => {
+    const actual = formatCo2(999_000_000_000_000);
+    const expected = '1 Gt';
+    expect(actual).toBe(expected);
+  });
```
Comment (madsnedergaard) [inline web/src/utils/formatting.test.ts:51]: Maybe put this number lower so that it is not rounded up to Gt - ideally it should return 0.X Gt

**Review by madsnedergaard — COMMENTED** [review]
Nice work, looks really promising 🙏 
A few comments, nothing major though

File: web/src/utils/formatting.ts [hunk 8]
```diff
@@ -17,7 +17,7 @@ export const formatPower = function (
   if (d == undefined || Number.isNaN(d)) {
     return d;
   }
-  const power = `${d3.format(`.${numberDigits}s`)(d * 1e6)}Wh`;
+  const power = `${d3.format(`.${numberDigits}s`)(d)}Wh`;
   return addSpaceBetweenNumberAndUnit(power);
```
[the remark this answers is not in the document]
Comment (madsnedergaard) [reply web/src/utils/formatting.ts:21]: Just checked, this is definitely needed - otherwise we show kWh instead of GWh.

Actual:
Screenshot 2023-09-07 at 13 08 01

Expected:
Screenshot 2023-09-07 at 13 10 02

File: web/src/utils/helpers.ts [hunk 9]
```diff
@@ -145,6 +145,6 @@ export function getNetExchange(
     return Number.NaN;
   }
   return displayByEmissions
-    ? round(zoneData.totalCo2NetExchange) // in CO₂eq
+    ? round(zoneData.totalCo2NetExchange ?? 0) // in CO₂eq
```
[the remark this answers is not in the document]
Comment (tonypls) [reply web/src/utils/helpers.ts]: Good point, I've reworked this to check if both import and export are missing and then return Number.NaN. Also I added a check to show "?" in the formatCo2 function if it receives a NaN.

File: web/src/features/charts/graphUtils.ts (hunk 3 above)
Comment (tonypls) [reply web/src/features/charts/graphUtils.ts:74]: Yeah I started writing getTotalEmissionOrElectricity when I figured this was probably time to split it up

File: web/src/features/charts/bar-breakdown/utils.ts [hunk 10]
```diff
@@ -44,7 +44,7 @@ export interface ProductionDataType {
   isStorage: boolean;
   storage: Maybe<number>;
   mode: ElectricityModeType;
-  tCo2eqPerMin: number;
+  gCo2eqPerHour: number;
```
[the remark this answers is not in the document]
Comment (tonypls) [reply web/src/features/charts/bar-breakdown/utils.ts]: Done, there's definitely a few more hour references around, we should make a separate PR to tidy up!

File: web/src/utils/formatting.test.ts (hunk 7 above)
Comment (tonypls) [reply web/src/utils/formatting.test.ts]: I think I've updated the tests now to cover all obvious edge cases, it's a little hard to know what 100k means now we've changed to grams as the base unit. Could you take a look at the formatting tests in their current state and give further feedback?

Comment (madsnedergaard) [reply web/src/utils/formatting.test.ts]: Fair point, will look through it all after we merge this PR :)

**Comment by tonypls** [comment]
Added some tests for Energy formatting some more naming polish

**Review by madsnedergaard — APPROVED** [review]
Great changes, thanks 😍
